src/services/layerManager.js

```diff
@@ -0,0 +1,252 @@
+goog.provide('ngeo.LayerManager');
+
+goog.require('ngeo');
+goog.require('ol.Collection');
+goog.require('ol.format.WMTSCapabilities');
+goog.require('ol.layer.Group');
+goog.require('ol.layer.Image');
+goog.require('ol.layer.Tile');
+goog.require('ol.source.ImageWMS');
+goog.require('ol.source.WMTS');
+
+
+
+/**
+ * @param {angular.$http} $http Angular http service.
+ * @constructor
+ */
+ngeo.LayerManager = function($http) {
+
+  /**
+   * @type {boolean}
+   * @private
+   */
+  this.initialized_ = false;
+
+  /**
+   * @type {ol.Map}
+   * @private
+   */
+  this.map_ = null;
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+};
+
+
+/**
+ * Init the service.
+ * @param {!ol.Map} map
+ * @export
+ */
+ngeo.LayerManager.prototype.init = function(map) {
+  if (!this.initialized_) {
+    this.map_ = map;
+    this.initialized_ = true;
+  }
+};
+
+
+/**
+ * Create and return a basic WMS layer with only a source URL and a dot
+ * separated layers names (see ol.source.ImageWMS).
+ * @param {string} sourceUrl The source url.
+ * @param {string} sourceLayersName A dot separated names string.
+ * @return {ol.layer.Image}
+ * @export
+ */
+ngeo.LayerManager.prototype.createBasicWMSLayer = function(sourceUrl,
+    sourceLayersName) {
+  var layer = new ol.layer.Image({
+    source: new ol.source.ImageWMS({
+      url: sourceUrl,
+      params: {'LAYERS': sourceLayersName}
+    })
+  });
+  return layer;
+};
+
+
+/**
+ * Create and return a WMTS layer. The layer source will be configured by
+ * the capabilities that are loaded from the given capabilitiesUrl.
+ * @param {string} capabilitiesUrl The getCapabilities url.
+ * @param {string} layerName The name of te layer.
+ * @return {ol.layer.Tile}
+ * @export
+ */
+ngeo.LayerManager.prototype.createWMTSLayerFromCapabilitites = function(
+    capabilitiesUrl, layerName) {
+  var parser = new ol.format.WMTSCapabilities();
+  var layer = new ol.layer.Tile();
+  this.http_.get(capabilitiesUrl).success(function(response) {
+    var result = parser.read(response);
+    var options = ol.source.WMTS.optionsFromCapabilities(result,
+        {layer: layerName, requestEncoding: 'REST'});
+    layer.setSource(new ol.source.WMTS(options));
+  }).error(function(response) {
+    //TODO handle error
+  });
+  return layer;
```

Comment:
Is it bug proof to have layer without source ?
What will you do if the capabilities fails, or if the source creation fails ?